Musigwa/RNLevelUp
# #166257426 Setup the testing environment for the project

#### What does this PR do?

- It sets up the testing environment
#### Description of Task to be completed?

- Add the necessary `jest` configurations
- Integrate the `Code coverage`, `Circle CI`, `Code climate`
- Integrate the `ESLint` and `Hound CI` for style checking.

#### How should this be manually tested?
- After cloning the repository, check out `ch-setup-test-environment-166257426` branch
- Checkout the `README.md` file for set up instructions
- Run `npm install` to install the project's dependencies
- Run `npm test` to runs all test scenarios and see the results in from the terminal
#### Any background context you want to provide?
- I'm using `ESLint` as the linter with `Hound CI` bot for automatic style guide checking.
- The `ESLint` configuration is extends the `Airbnb` react/react native style guide with eslint-prettier integration plugins.
#### What are the relevant pivotal tracker stories?
- #166257426
